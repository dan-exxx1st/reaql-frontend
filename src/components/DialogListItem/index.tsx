import React, { FC } from 'react';
import StyledDialogListItem from './style';

const DialogListItem: FC = () => {
	return <StyledDialogListItem>Dialog list item</StyledDialogListItem>;
};

export default DialogListItem;
