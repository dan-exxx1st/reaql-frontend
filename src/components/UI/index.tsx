export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Circle } from './Circle';
export { default as SkeletonLoader } from './SkeletonLoader';
export { default as TextField } from './TextField';
export { default as Typography } from './Typography';
export { default as Wrapper } from './Wrapper';
export { default as CheckMark } from './CheckMark';
export { default as Icon } from './Icon';
export { default as CheckBox } from './CheckBox';
export { default as SystemMessage } from './SystemMessage';
export { default as Spinner } from './Spinner';
