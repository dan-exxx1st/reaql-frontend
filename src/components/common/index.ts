export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
