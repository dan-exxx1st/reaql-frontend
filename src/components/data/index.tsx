export { default as MessageListWithData } from './MessageListWithData';
export { default as DialogHeaderWithData } from './DialogHeaderWithData';
