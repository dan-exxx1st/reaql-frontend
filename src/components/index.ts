export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
