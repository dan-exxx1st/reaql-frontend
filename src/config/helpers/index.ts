import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
