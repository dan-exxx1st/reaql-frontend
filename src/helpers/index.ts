import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
