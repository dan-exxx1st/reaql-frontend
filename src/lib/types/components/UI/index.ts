import { MouseEventHandler } from 'react';
import Theme from 'helpers/styled';
import { ColorType, IconName } from 'lib/types';
import { TypographyVariantType, TypographyComponentType } from 'lib/types/ui';

export interface IAvatarProps {
    src?: HTMLImageElement['src'];
    alt?: string;
    className?: string;
}

export interface IButtonProps {
    className?: string;
    theme: typeof Theme;
    onClick?: MouseEventHandler;
    variant: 'filled' | 'outlined';
    color: ColorType;
}

export interface ICheckMarkProps {
    isChecked: boolean;
    isDouble: boolean;
}

export interface ICircleProps {
    color: ColorType;
    iconName?: IconName;
    text?: string | number;
    width?: string;
    height?: string;
    cursor?: string;
    className?: string;
    onClick?: () => void;
}

export interface IIconProps {
    iconName: IconName;
    className?: string;
}

export interface ISkeletonLoaderProps {
    variant?: 'avatar' | 'block';
    width?: string;
    height?: string;
    className?: string;
}

export interface ITextFieldProps {
    onChange?: (value: string) => void;
    value?: string;
    className?: string;
    fieldSize: 'small' | 'medium' | 'large';
    color: ColorType;
    icon?: IconName;
    width?: string;
    height?: string;
    wrapperWidth?: string;
    placeholder?: string;
    type?: string;
}

export interface ITypographyProps {
    className?: string;
    variant: TypographyVariantType;
    component?: TypographyComponentType;
    color?: string;
    theme: typeof Theme;
}

export type { IWrapperProps } from './WrapperTypes';

export interface ICheckBoxProps {
    onClick?: () => void;
    text?: string;
}
