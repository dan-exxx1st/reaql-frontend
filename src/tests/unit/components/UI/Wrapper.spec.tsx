import React from 'react';
import { shallowWithTheme, mountWithTheme } from 'tests/helpers/withTheme';
import { Wrapper } from 'components/UI';

describe('<Wrapper />', () => {
    describe('Snapshots', () => {
        it('Should render a wrapper correctly with default styles', () => {
            const wrapper = shallowWithTheme(
                <Wrapper>Correctly wraper</Wrapper>
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('Should have a correctly style rules', () => {
        const wrapper = mountWithTheme(
            <Wrapper
                flexDirection="column"
                justifyContent="center"
                alignContent="center"
                alignItems="center"
                margin="5px"
                padding="15px"
            >
                Correctly wraper
            </Wrapper>
        );

        expect(wrapper).toHaveStyleRule('flex-direction', 'column');
    });

    it('Should have a correctly default style rules', () => {
        const wrapper = mountWithTheme(<Wrapper>Correctly wraper</Wrapper>);

        expect(wrapper).toHaveStyleRule('flex-direction', 'row');
        expect(wrapper).toHaveStyleRule('justify-content', 'flex-start');
        expect(wrapper).toHaveStyleRule('align-content', 'flex-start');
        expect(wrapper).toHaveStyleRule('align-items', 'flex-start');
    });
});
