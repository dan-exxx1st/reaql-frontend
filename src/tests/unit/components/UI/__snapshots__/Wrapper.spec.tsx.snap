// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> Snapshots Should render a wrapper correctly with default styles 1`] = `
<ContextProvider
  value={
    Object {
      "dc": "#252525",
      "dgc": "#A5A5A5",
      "dsc": "#555555",
      "ff": "Roboto",
      "fontSizes": Object {
        "body1": "16px",
        "body2": "14px;",
        "caption1": "12px",
        "caption2": "12px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
      },
      "fontWeights": Object {
        "body1": "normal",
        "body2": "normal",
        "caption1": "normal",
        "caption2": "bold",
        "h1": "300",
        "h2": "300",
        "h3": "normal",
        "h4": "normal",
        "h5": "normal",
        "h6": "500",
      },
      "formBoxShadow": "5px 5px 15px rgba(37, 37, 37, 0.1)",
      "gc": "#E5E5E5",
      "lbc": "#E8F3F6",
      "letterSpacing": Object {
        "body1": "0.5px",
        "body2": "0.25px",
        "caption1": "0.4px",
        "caption2": "0.4px",
        "h1": "-1.5px",
        "h2": "-0.5px",
        "h3": "0.25px",
        "h6": "0.15px",
      },
      "lineHeight": Object {
        "body1": "19px",
        "caption1": "14px",
        "caption2": "14px",
        "h1": "112px",
        "h2": "70px",
        "h3": "56px",
        "h4": "40px",
        "h5": "28px",
        "h6": "23px",
      },
      "pc": "#00ADFF",
      "plc": "#A4E2FF",
      "sc": "#E6E6E6",
      "wc": "#FAFAFA",
    }
  }
>
  <Wrapper>
    Correctly wraper
  </Wrapper>
</ContextProvider>
`;
